realm/realm-library/src/objectServer/java/io/realm/RealmAppConfiguration.java

```diff
@@ -219,25 +294,97 @@ public Builder appVersion(String appVersion) {
         /**
          * FIXME
          *
-         * @param errorHandler
+         * @param time
+         * @param unit
          * @return
          */
-        public Builder defaultSessionErrorHandler(@Nullable SyncSession.ErrorHandler errorHandler) {
-            // FIXME checks
-            this.defaultErrorHandler = errorHandler;
+        public Builder requestTimeout(long time, TimeUnit unit) {
+            if (time < 1) {
+                throw new IllegalStateException("A timeout above 0 is required: " + time);
+            }
+            Util.checkNull(unit, "unit");
+            this.requestTimeoutMs = TimeUnit.MICROSECONDS.convert(time, unit);
             return this;
         }
 
         /**
-         * FIXME
+         * Sets the name of the HTTP header used to send authorization data in when making requests to
+         * MongoDB Realm. The MongoDB server or firewall must have been configured to expect a
+         * custom authorization header.
+         * <p>
+         * The default authorization header is named "Authorization".
          *
-         * @param time
-         * @param unit
+         * @param headerName name of the header.
+         * @throws IllegalArgumentException if a null or empty header is provided.
+         * @see <a href="https://docs.realm.io/platform/guides/learn-realm-sync-and-integrate-with-a-proxy#adding-a-custom-proxy">Adding a custom proxy</a>
+         */
+        public Builder authorizationHeaderName(String headerName) {
+            Util.checkEmpty(headerName, "headerName");
+            this.autorizationHeaderName = headerName;
+            return this;
+        }
+
+        /**
+         * Adds an extra HTTP header to append to every request to a Realm Object Server.
+         *
+         * @param headerName the name of the header.
+         * @param headerValue the value of header.
+         * @throws IllegalArgumentException if a non-empty {@code headerName} is provided or a null {@code headerValue}.
+         */
+        public Builder addCustomRequestHeader(String headerName, String headerValue) {
+            Util.checkEmpty(headerName, "headerName");
+            Util.checkNull(headerValue, "headerValue");
+            customHeaders.put(headerName, headerValue);
+            return this;
+        }
+
+        /**
+         * Adds extra HTTP headers to append to every request to a Realm Object Server.
+         *
+         * @param headers map of (headerName, headerValue) pairs.
+         * @throws IllegalArgumentException If any of the headers provided are illegal.
+         */
+        public Builder addCustomRequestHeaders(@Nullable Map<String, String> headers) {
+            if (headers != null) {
+                customHeaders.putAll(headers);
```

Comment:
Not really any check for illegal headers

Reply:
Good point. It mostly isn't there because I would expect the lower layers to validate as needed, but we need to add tests for it. I'll add a FIXME

Reply:
Added a FIXME in the tests